<launch>
    <!-- Specify the image input top from the camera -->
    <arg name="input_image" value="/external_camera_overhead/ec_depth/image_raw"/>
    <arg name="input_edge_type" default="1"/>
	<arg name="input_max_corners" default="100"/>
	<arg name="input_canny_low_threshold" default="47"/>
	<arg name="input_threshold" default="142"/>

    <!--load edge detection to locate edges publishes image to /[node_name]/image [default node name: edge_detection_filter]-->
    <!-- <include file="$(find opencv_apps)/launch/edge_detection.launch">
        <arg name="node_name" value="edge_detection_filter"/>
        <arg name="image" value="$(arg input_image)"/>
        <arg name="edge_type" value="$(arg input_edge_type)"/>
		<arg name="debug_view" value="true"/>
    </include> -->
	<include file="$(find opencv_apps)/launch/goodfeature_track.launch">
        <arg name="node_name" value="edge_detection_filter"/>
        <arg name="image" value="$(arg input_image)"/>
		<arg name="max_corners" value="$(arg input_max_corners)" />
		<arg name="debug_view" value="true" />
    </include>
	<!-- <include file="$(find opencv_apps)/launch/hough_lines.launch">
        <arg name="node_name" value="edge_detection_filter"/>
        <arg name="image" value="$(arg input_image)"/>
		<arg name="threshold" value="$(arg input_threshold)"/>
		<arg name="debug_view" value="true" />
		<arg name="hough_type" value="0" />
		<arg name="rho" value="1.0" />
		<arg name="theta" value="0.5" />
    </include> -->

    <!--calculate image moments for contours in input image.
		Using output of rgb colour filter, this gives coordinates of 
			colour patches extracted from filter.

		publishes image to /[node_name]/image
		moments to /[node_name]/moments
			default node name: contour_moments
	-->
	<include file="$(find opencv_apps)/launch/contour_moments.launch">
		<arg name="node_name" value="contour_moments_edge_detection" />
		<arg name="image" value="/edge_detection_filter/image"/>
		<arg name="debug_view" value="true" />
		<arg name="canny_low_threshold" value="$(arg input_canny_low_threshold)" />
	</include>
	<!-- <include file="$(find opencv_apps)/launch/general_contours.launch">
		<arg name="node_name" value="contour_moments_edge_detection" />
		<arg name="image" value="/edge_detection_filter/image"/>
		<arg name="debug_view" value="true" />
		<arg name="threshold" value="$(arg input_canny_low_threshold)" />
	</include> -->

</launch>